scene/debugger/scene_debugger.cpp

```diff
@@ -173,8 +177,55 @@ Error SceneDebugger::parse_message(void *p_user, const String &p_msg, const Arra
 		}
 		scene_tree->get_root()->set_camera_3d_override_transform(transform);
 		runtime_node_select->_queue_selection_update();
+	} else if (p_msg == "add_preview_nodes_to_current_3d_scene") {
+		ERR_FAIL_COND_V(p_args.size() < 6, ERR_INVALID_DATA);
+		const bool should_add_world_environment = p_args[0];
+		const bool should_add_directional_light_3d = p_args[1];
+		const bool should_add_camera_3d = p_args[2];
+
+		List<String> added_node_names;
+		if (should_add_world_environment) {
+			WorldEnvironment *world_environment = Object::cast_to<WorldEnvironment>(VariantUtilityFunctions::bytes_to_var_with_objects(p_args[3]));
+			added_node_names.push_back(world_environment->get_class_name());
+			ERR_FAIL_COND_V(world_environment == nullptr, ERR_INVALID_DATA);
+
+			scene_tree->get_root()->add_child(world_environment, false, Node::INTERNAL_MODE_BACK);
+			world_environment->set_name("WorldEnvironmentPreview");
+		}
+
+		if (should_add_directional_light_3d) {
+			DirectionalLight3D *sun = Object::cast_to<DirectionalLight3D>(VariantUtilityFunctions::bytes_to_var_with_objects(p_args[4]));
+			added_node_names.push_back(sun->get_class_name());
+			ERR_FAIL_COND_V(sun == nullptr, ERR_INVALID_DATA);
+
+			scene_tree->get_root()->add_child(sun, false, Node::INTERNAL_MODE_BACK);
+			sun->set_name("SunPreview");
+		}
+
+		if (should_add_camera_3d) {
+			Camera3D *camera_3d = Object::cast_to<Camera3D>(VariantUtilityFunctions::bytes_to_var_with_objects(p_args[5]));
+			added_node_names.push_back(camera_3d->get_class_name());
+			ERR_FAIL_COND_V(camera_3d == nullptr, ERR_INVALID_DATA);
+
+			scene_tree->get_root()->add_child(camera_3d, false, Node::INTERNAL_MODE_BACK);
+			camera_3d->set_name("Camera3DPreview");
+			camera_3d->make_current();
+		}
+
+		if (added_node_names.size() > 0) {
+			if (added_node_names.size() == 1) {
+				WARN_PRINT_ED(vformat(TTR("Current scene was run without a %s node. Adding %d preview node to the scene. This will not carry over to the exported project. \nYou can disable this by changing `editor/run/automatically_add_preview_nodes_when_running_single_3d_scene` in the Project Settings."), added_node_names.get(0), added_node_names.size()));
+			} else if (added_node_names.size() == 2) {
+				WARN_PRINT_ED(vformat(TTR("Current scene was run without %s and %s nodes. Adding %d preview nodes to the scene. This will not carry over to the exported project. \nYou can disable this by changing `editor/run/automatically_add_preview_nodes_when_running_single_3d_scene` in the Project Settings."), added_node_names.get(0), added_node_names.get(1), added_node_names.size()));
```

Comment:
```suggestion
				WARN_PRINT_ED(vformat(TTR("Current scene was run without %s and %s nodes. Adding %d preview nodes to the scene. This will not carry over to the exported project. \nYou can disable this by changing `editor/run/automatically_add_preview_nodes_when_running_single_3d_scene` in the Project Settings."), added_node_names.front()->get(), added_node_names.get(1), added_node_names.size()));
```